packages/core/src/cache/__test__/cache.test.ts

```diff
@@ -398,4 +404,146 @@ describe('cache', () => {
     expect(scope.getState(query.$data)).toEqual({ step: 2 });
     expect(scope.getState(query.$stale)).toBeFalsy();
   });
+
+  test('do not cache data if disabled', async () => {
+    const $enabled = createStore(true);
+
+    let index = 1;
+    const handler = vi.fn(async (p: void) =>
+      setTimeout(1000).then(() => index++)
+    );
+    const query = withFactory({
+      fn: () =>
+        createQuery({
+          handler,
+        }),
+      sid: '1',
+    });
+
+    cache(query, {
+      /**
+       * It doesn't make sense to use cache without staleAfter,
+       * bcs it will be always stale,
+       * so:
+       * - (using query.start) it will be always refetched from remote source
+       * - (using query.refresh) it will allways be skipped due to freshness(provided from remote source)
+       *
+       */
+      staleAfter: '1h',
+
+      enabled: $enabled,
+    });
+
+    const scope = fork();
+
+    // Load from remote data-source
+    {
+      allSettled(query.start, { scope });
+      // But wait for next tick becuase of async adapter's nature
+      await setTimeout(1);
+      await allSettled(scope);
+
+      expect(scope.getState(query.$data)).toEqual(1);
+    }
+
+    // query.start should read from cache
+    {
+      allSettled(query.start, { scope });
```

Comment:
It feels wrong, because from my point of view: 
- `query.start` should be `always use remote source` (like force fetching new data)
- `query.refresh` should be `go to remote source if cache is stale` (current query.start implementation)

It all points to one thing: data from remoteSource should be `stale: true` by default rather than `stale: false` https://github.com/igorkamyshev/farfetched/blob/3ec5631e37fa03b402bcd08a93d6de2fb49ffc06/packages/core/src/remote_operation/create_remote_operation.ts#L100

Am I missing something here?

Reply:
```
let count = 0;
const query = createQuery({ 
 effect: () => {
    count += 1;
    return count;
 } 
});
const scope = fork();
await allSettled(query.refresh, { scope });
await allSettled(query.refresh, { scope });
expect(count).toBe(2); // fails bcs it first call cached (but i didnt attach any cache)
```

Reply:
It is not 🤔 Your original assumption is not synced with the current behavior. I believe we should not change while adding such a minor change.